packages/protocol/contracts/L1/verifiers/SgxVerifier.sol

```diff
@@ -161,13 +176,25 @@ contract SgxVerifier is EssentialContract, IVerifier {
         );
     }
 
-    function _addInstances(address[] calldata _instances) private returns (uint256[] memory ids) {
+    function _addInstances(
+        address[] memory _instances,
+        bool instantValid
+    )
+        private
+        returns (uint256[] memory ids)
+    {
         ids = new uint256[](_instances.length);
 
+        uint64 validSince = uint64(block.timestamp);
+
+        if (!instantValid) {
+            validSince += INSTANCE_VALIDITY_DELAY;
+        }
+
         for (uint256 i; i < _instances.length; ++i) {
             if (_instances[i] == address(0)) revert SGX_INVALID_INSTANCE();
 
-            instances[nextInstanceId] = Instance(_instances[i], uint64(block.timestamp));
+            instances[nextInstanceId] = Instance(_instances[i], validSince);
             ids[i] = nextInstanceId;
 
             emit InstanceAdded(nextInstanceId, _instances[i], address(0), block.timestamp);
```

Comment:
Should the event also use `validSince` instead of `block.timestamp`. Seems more useful.